scripts/plugins/extauth-hook-AD.py

```diff
@@ -233,6 +223,69 @@ def _install(self):
                 os.remove(self._file_path)
 
 
+class UsersList(DynamicPam):
+    #pylint: disable=too-few-public-methods
+    """Class manage users which permit pool admin ssh"""
+
+    def __init__(self, session, arg, ad_enabled=True):
+        super(UsersList, self).__init__(HCP_USERS, session, arg, ad_enabled)
+
+    def _match_subject(self, subject_rec):
+        return subject_rec["other_config"]["subject-is-group"] != "true"
+
+    def _add_upn(self, subject_rec):
+        sep = "@"
+        try:
+            upn = subject_rec["other_config"]["subject-upn"]
+            user, domain = upn.split(sep)
+            if self._backend == ADBackend.BD_PBIS:
+                # PBIS convert domain to UPPER case, we revert it back
+                domain = domain.lower()
+            self._lines.append("{}{}{}".format(user, sep, domain))
+        except KeyError:
+            logger.info("subject does not have upn %s", subject_rec)
+        except ValueError:
+            logger.info("UPN format is not right %s", upn)
+
+    def _add_subject(self, subject_rec):
+        try:
+            sid = subject_rec['subject_identifier']
+            name = subject_rec["other_config"]["subject-name"]
+            formatted_name = self._format_item(name)
+            logger.debug("Permit user %s with sid %s", formatted_name, sid)
+            self._lines.append(formatted_name)
+            # If ssh key is permittd in authorized_keys,
+            # The original name is compared, add UPN and original name
+            if self._backend == ADBackend.BD_PBIS and name != formatted_name:
+                self._lines.append(name)
+            self._add_upn(subject_rec)
+        # pylint: disable=broad-except
+        except Exception as exp:
+            logger.warning("Failed to add user %s: %s", subject_rec, str(exp))
+
+
+class GroupsList(DynamicPam):
+    #pylint: disable=too-few-public-methods
+    """Class manage groups which permit pool admin ssh"""
+
+    def __init__(self, session, arg, ad_enabled=True):
+        super(GroupsList, self).__init__(HCP_GROUPS, session, arg, ad_enabled)
+
+    def _match_subject(self, subject_rec):
+        return subject_rec["other_config"]["subject-is-group"] == "true"
```

Comment:
as above, what if this key is missing from other-config? it'll raise exception and crash, would be better to check that 'subject-is-group' is in other-config, and if not then return false here. Also is `other_config` spelled correctly here, shouldn't it be `other-config`?